### PR TITLE
Add bitlength parameter

### PR DESCRIPTION
**Pull request by giuseppeg88**
Some new 433 MHz RF switches use codes with 32 bit (against the usual 24 bit): this commit allows user to specify the code length.
In addition, a verification is implemented that automatically sets the length to 32 bit when a user tries to send a code greater than 24 bits.

### CODE REVIEW

File: rpi_rf/rpi_rf.py [hunk 1]
```diff
@@ -87,27 +87,40 @@ def disable_tx(self):
...
         return True
 
-    def tx_code(self, code, tx_proto=None, tx_pulselength=None):
+    def tx_code(self, code, tx_proto=None, tx_pulselength=None, tx_length=None):
         """
         Send a decimal code.
 
-        Optionally set protocol and pulselength.
-        When none given reset to default protocol and pulselength.
+        Optionally set protocol, pulselength and (code) lenght.
```
Comment (milaq) [inline rpi_rf/rpi_rf.py]: typo ("length")

File: rpi_rf/rpi_rf.py [hunk 2]
```diff
@@ -87,27 +87,40 @@ def disable_tx(self):
...
+        When none given reset to default protocol and pulselength and set to 24 bit (code) lenght.
         """
         if tx_proto:
             self.tx_proto = tx_proto
         else:
             self.tx_proto = 1
+
```
Comment (milaq) [inline rpi_rf/rpi_rf.py]: why is that line removed?

File: rpi_rf/rpi_rf.py [hunk 3]
```diff
@@ -87,27 +87,40 @@ def disable_tx(self):
...
 
-    def tx_code(self, code, tx_proto=None, tx_pulselength=None):
+    def tx_code(self, code, tx_proto=None, tx_pulselength=None, tx_length=None):
         """
         Send a decimal code.
 
-        Optionally set protocol and pulselength.
-        When none given reset to default protocol and pulselength.
+        Optionally set protocol, pulselength and (code) lenght.
+        When none given reset to default protocol and pulselength and set to 24 bit (code) lenght.
```
Comment (milaq) [inline rpi_rf/rpi_rf.py]: same typo here

File: rpi_rf/rpi_rf.py (hunk 2 above)
Comment (giuseppeg88) [reply rpi_rf/rpi_rf.py]: there is no reason

**Comment by milaq** [comment]
So,
I made some minor changes to your code (see https://github.com/milaq/rpi-rf/commit/8138582d8a291a59fd15f8a126c5e0143e1b8c17):
 * Switch conditional default in tx_code() to 24 bits
 * Allow (very rare) exact 24 bit code size to be sent as 24 bit
 * Refactor wording
 * Do not override tx_length in tx_bin(). The global paramater gets set in tx_code already

Next time, please be sure to make clean PRs (squash your changes, make changes only on the topic of the commit, meaningful commit messages, etc.).

Please have a look at the changes, test it if you have the time and let me know if everything is on order.
I'll keep this PR open until I hear back from you.
Thanks again for your work :+1:

**Comment by giuseppeg88** [comment]
I have carried out tests with codes of different lengths and I have not found any problems. Thank you

**Comment by milaq** [comment]
Nice, thanks for your feedback
